common/src/main/java/com/wynntils/screens/maps/PoiManagementScreen.java

```diff
@@ -11,40 +11,84 @@
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.EnumMap;
 import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
 import java.util.stream.Stream;
 import net.minecraft.ChatFormatting;
 import net.minecraft.client.gui.GuiGraphics;
 import net.minecraft.client.gui.components.AbstractWidget;
 import net.minecraft.client.gui.components.Button;
 import net.minecraft.client.gui.components.Tooltip;
+import net.minecraft.client.gui.components.events.GuiEventListener;
 import net.minecraft.client.gui.screens.Screen;
 import net.minecraft.client.resources.language.I18n;
 import net.minecraft.network.chat.Component;
 
-public final class PoiManagementScreen extends WynntilsScreen {
-    private final MainMapScreen oldMapScreen;
-
-    private Button nextButton;
-    private Button previousButton;
-    private Button undoDeleteButton;
+public final class PoiManagementScreen extends WynntilsScreen implements TextboxScreen {
+    private static final float GRID_DIVISIONS = 64.0f;
+    private static final int GRID_ROWS_PER_PAGE = 43;
+    private static final int HEADER_HEIGHT = 12;
 
-    private List<CustomPoi> waypoints;
-    private int pageHeight;
-    private int currentPage;
+    private final List<AbstractWidget> poiManagerWidgets = new ArrayList<>();
     private final List<CustomPoi> deletedPois = new ArrayList<>();
     private final List<Integer> deletedIndexes = new ArrayList<>();
-    private final List<AbstractWidget> poiManagerWidgets = new ArrayList<>();
+    private final MainMapScreen oldMapScreen;
+
+    private boolean draggingScroll = false;
+    private boolean selectionMode = false;
+    private Button deleteSelectedButton;
+    private Button deselectAllButton;
+    private Button downButton;
+    private Button exportButton;
+    private Button filterButton;
+    private Button selectAllButton;
+    private Button undoDeleteButton;
+    private Button upButton;
+    private float backgroundHeight;
+    private float backgroundWidth;
+    private float backgroundX;
+    private float backgroundY;
+    private float dividedHeight;
+    private float dividedWidth;
+    private float scrollButtonHeight;
+    private float scrollButtonRenderX;
+    private float scrollButtonRenderY;
+    private int maxPoisToDisplay;
+    private int scrollAreaHeight;
+    private int scrollOffset = 0;
+    private List<CustomPoi> selectedPois = new ArrayList<>();
+    private List<CustomPoi> pois;
+    private Map<Texture, Boolean> filteredIcons = new EnumMap<>(Texture.class);
+    private PoiSortButton activeSortButton;
+    private PoiSortButton iconSortButton;
+    private PoiSortButton nameSortButton;
+    private PoiSortButton xSortButton;
+    private PoiSortButton ySortButton;
+    private PoiSortButton zSortButton;
+    private PoiSortOrder sortOrder;
+    private TextInputBoxWidget focusedTextInput;
+    private TextInputBoxWidget searchInput;
 
```

Comment:
Perhaps keep some kind of groupping here?

I like the approach I used in the `TradeMarketHolder` class in the custom TM PR. Basically just comment regions (but not actual // region ...)